easybuild/tools/github.py

```diff
@@ -1331,6 +1331,9 @@ def merge_pr(pr):
     force = build_option('force')
     dry_run = build_option('dry_run') or build_option('extended_dry_run')
 
+    if pr_data['merged'] and not dry_run:
```

Comment:
Should we also take into account closed PRs?

Also, why doesn't this make the `merge_pr` test in `test/framework/options.py` fail?

Reply:
> Should we also take into account closed PRs?

will do

> Also, why doesn't this make the `merge_pr` test in `test/framework/options.py` fail?

that's why I added the `not dry_run`

Reply:
Should we update the test so it tries `--merge-pr` without dry run on a merged/closed PR to cover this change in the tests?